skeleton/src/main/java/org/killbill/commons/skeleton/metrics/TimedResourceInterceptor.java

```diff
@@ -29,33 +43,133 @@
  */
 public class TimedResourceInterceptor implements MethodInterceptor {
 
-    private final TimedResourceMetric timer;
+    private final Provider<GuiceContainer> jerseyContainer;
+    private final Provider<MetricRegistry> metricRegistry;
+    private final String resourcePath;
```

Comment:
Any reason why we don't keep track of `MetricRegistry` and `GuiceContainer ` instead of `Provider<GuiceContainer>` and `Provider<MetricRegistry>` ?